src/lexer/tiger.lex

```diff
@@ -1,7 +1,10 @@
 type pos = int
 type lexresult = Token.token
 
-fun eof() = Token.EOF
+fun eof() =
+    (if SrcString.isBuildingString() then raise SrcString.StringNotClosed(Newline.getPos(SrcString.getStartPos()), Newline.getLine(SrcString.getStartPos()))
+    else if not (SrcComment.closed()) then raise SrcComment.CommentNotClosed(Newline.getPos(hd(SrcComment.getComments())), Newline.getLine(hd(SrcComment.getComments())))
```

Comment:
We shouldn't be throwing exceptions, instead we should be using the ErrorMsg struct's error function to report issues.